### ISSUE TITLE
Failed to archive link: KeyError: 'base_url'

### ISSUE THREAD

**Issue by raman325** [pirate/ArchiveBox#157]
**Describe the bug**
On every URL being archived, all archive steps seem to be successful but then I get this error before moving on to the next URL. I am using Pinboard's GET posts API which appears to give the same output as exporting from Pinboard directly: `curl "https://api.pinboard.in/v1/posts/all?auth_token=<user>:<token>"` The end result is that the output.html of the page works, but the index.html does not (index.json has content though).

**Steps to reproduce**
1. Use default conf file
2. Run CURL command using a valid user and token and pipe it to archive
3. See error

**Screenshots or log output**
```
[+] [2019-03-02 18:12:53] "https://www.thesimpledollar.com/teaching-children-about-compound-interest/"
    https://www.thesimpledollar.com/teaching-children-about-compound-interest/
    > /data/archive/1551550347.0 (new)
      > favicon
      > title
      > wget
      > pdf
      > screenshot
      > dom
      > archive_org
      > git
      > media
      √ index.json
      √ index.html
    ! Failed to archive link: KeyError: 'base_url'
```

**Software versions (please complete the following information):**
 - ArchiveBox version: master (63e6ed8)
 - Python version: Latest version on Debian Stretch (node:11-slim)
 - OS: Synology DSM 6.2.1-23824 Update 6
 - Chrome version: Latest google-chrome-unstable version on Debian Stretch (node:11-slim)

**Comment by pirate** [comment]
Ah sorry, this was a bug from a recent refactor.  Just fixed it in cb573bb

Pull the latest master and give it a try, comment back if it doesn't work and I'll reopen the ticket.

**Comment by raman325** [comment]
Looks like that did the trick, thanks!
